packages/connector-types/src/index.ts

```diff
@@ -0,0 +1,120 @@
+import { Languages } from '@logto/phrases';
+import { ArbitraryObject, Connector, ConnectorType } from '@logto/schemas';
+
+import { Response } from './aliyun';
+
+export * from './aliyun';
+
+export interface ConnectorMetadata {
+  id: string;
+  type: ConnectorType;
+  name: Record<Languages, string>;
+  logo: string;
+  description: Record<Languages, string>;
+  readme: string;
+  configTemplate: string;
+}
+
+export interface ConnectorDTO extends Connector {
+  metadata: ConnectorMetadata;
+}
+
+export enum ConnectorErrorCodes {
+  General,
+  InsufficientRequestParameters,
+  InvalidConfig,
+  InvalidResponse,
+  TemplateNotFound,
+  SocialAuthCodeInvalid,
+  SocialAccessTokenInvalid,
+}
+
+export class ConnectorError extends Error {
+  public code: ConnectorErrorCodes;
+
+  constructor(code: ConnectorErrorCodes, message?: string) {
+    super(message);
+    this.code = code;
+  }
+}
+
+export type EmailMessageTypes = {
+  SignIn: {
+    code: string;
+  };
+  Register: {
+    code: string;
+  };
+  ForgotPassword: {
+    code: string;
+  };
+  Test: Record<string, unknown>;
+};
+
+type SmsMessageTypes = EmailMessageTypes;
+
+export type SendEmailResponse = { EnvId: string; RequestId: string };
+
+export type SendSmsResponse = { BizId: string; Code: string; Message: string; RequestId: string };
+
+export type EmailSendMessageFunction<
+  T1 extends ArbitraryObject = ArbitraryObject,
```

Comment:
will it be better to use meaningful names in generic when it has more than one type?